react/src/components/blocks/SearchForm/SearchForm.js

```diff
@@ -1,5 +1,56 @@
+import React, { useRef, useState, useEffect } from 'react'
+
+// ui
+import SettingsIcon from '@/ui/icons/SettingsIcon'
+import AutoCompleteInput from '@/ui/Fields/AutoCompleteInput'
+import Controls from './Controls'
+
+// image import
+import GoogleLogo from '../../../static/images/google_logo.png'
+
+// styles
+import styles from './SearchForm.module.css'
+
 function SearchForm() {
-  return <p>Search Form component</p>
+  const [down, setDown] = useState(false)
+  const prevScrollY = useRef(0)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      const currentScrollY = window.scrollY
+
+      prevScrollY.current = currentScrollY
+
+      setDown(currentScrollY > 300)
+    }
+
+    window.addEventListener('scroll', handleScroll, { passive: true })
+
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [])
```

Comment:
create custom hook

```
useScrollPosition.js

const useScrollPosition = () => {
  const [scrollPosition, setScrollPosition] = useState(0);

  useEffect(() => {
    const updatePosition = () => {
      setScrollPosition(window.pageYOffset);
    }
    window.addEventListener("scroll", updatePosition);
    updatePosition();
    return () => window.removeEventListener("scroll", updatePosition);
  }, []);

  return scrollPosition;
};

export default useScrollPosition;
```

Refer https://designcode.io/react-hooks-handbook-usescrollposition-hook

Reply:
That code was not needed. So removed it. This could be ignored.